app/helpers/template_helpers.py

```diff
@@ -61,7 +61,12 @@ def context(self) -> dict[str, Any]:
             "account_service_todo_url": self._survey_config.account_service_todo_url,
             "contact_us_url": self._survey_config.contact_us_url,
             "thank_you_url": url_for("post_submission.get_thank_you"),
-            "cookie_settings_url": self._survey_config.cookie_settings_url,
+            "cookie_settings_url": self._survey_config.cookie_settings_url
+            if cookie_session
```

Comment:
Also, it should be using `self._survey_type`, currently if you refresh the page you will still have a cookie session hence the banner shows. Indicates a missing test, I would add a functional test for this, also add a page refresh and assert to show it is still not there.

Reply:
Ok, this logic fixed now, thanks!

Reply:
Also added whole new spec for cookie banner testing with the tests requested bar the change settings link check cause we don't have a html identifier for it in our template.